brotab/inout.py

```diff
@@ -9,35 +9,41 @@
 from subprocess import check_call, CalledProcessError
 
 
-def is_port_accepting_connections(port):
+def is_port_accepting_connections(port, host='127.0.0.1'):
     s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
     s.settimeout(0.100)
-    result = s.connect_ex(('127.0.0.1', port))
+    result = s.connect_ex((host, port))
     s.close()
     return result == 0
 
 
 def save_tabs_to_file(tabs, filename):
-    with open(filename, 'w') as file_:
+    with open(filename, 'w', encoding='utf-8') as file_:
         file_.write('\n'.join(tabs))
 
 
 def load_tabs_from_file(filename):
-    with open(filename) as file_:
-        return [line.strip() for line in file_.readlines()]
+    with open(filename, encoding='utf-8') as file_:
+        Lines = [line.strip() for line in file_.readlines()]
+    if os.path.exists(filename):
+        os.remove(filename)
+    return Lines
 
 
 def edit_tabs_in_editor(tabs_before):
     with NamedTemporaryFile() as file_:
-        save_tabs_to_file(tabs_before, file_.name)
+        file_name=file_.name
+        file_.close()
```

Comment:
@naeloob why is this necessary?

Reply:
Because NamedTemporaryFiler lefts the file opened and, almost in windows, when save_tabs_to_file try to open it, it fails. That way, we get the temporary filename and close the file.
I've seen an error en my line 41 : 
`check_call([os.environ.get('EDITOR', Editor), file_.name])`
must be : 
`check_call([os.environ.get('EDITOR', Editor), file_name])`